design/pedibus/override/templates/adults/webpage_adults.tpl

```diff
@@ -0,0 +1,130 @@
+{def $__TITLE = $node.name|wash()}
+{def $__LIMIT = ezini( 'IlPedibus', 'adults_page_limit', 'ilpedibus.ini' )}
+{def $__ADULTS = fetch(
+                        'content',
+                        'list',
+                        hash(
+                            'parent_node_id', $node.node_id,
+                            'class_identifier', 'adulto',
+                            'sort_by', array( 'priority', true() )
+                        )
+                )
+}
+
+{if $node.data_map.short_name.has_content}
+    {set $__TITLE = $node.data_map.short_name.content|wash()}
+{/if}
+<div class="container-fluid main_cage row_main_title adults">
+    <div class="row">
+        <div class="col-xs-12">
+            <h1>{$__TITLE}</h1>
+        </div>
+    </div>
+</div>
+
+{*{$__ADULTS|attribute(show)}*}
+
+{if $__ADULTS|count()}
+    {def $current_user = fetch( 'user', 'current_user' )}
+    {def $__LINES_FILTER = array()}
+    {if $current_user.is_logged_in}
+        {if $current_user.contentobject.class_identifier|eq("user_pedibus")}
+            {foreach $current_user.contentobject.data_map.lines_enable_to_read.content.relation_list as $__LINE_REF}
+                {set $__LINES_FILTER = $__LINES_FILTER|append($__LINE_REF.node_id)}
+            {/foreach}
+        {/if}
+    {/if}
+
+{*{$__LINES_FILTER|attribute(show)}*}
+
+{*<p>
+    A == {$current_user.is_logged_in|attribute(show)}
+</p>
+<p>
+    B == {$current_user.contentobject.name|attribute(show)}
+</p>*}
+
+    {cache-block subtree_expiry=$__ADULTS keys=concat($node.name|wash(),$node.node_id,"adults",$view_parameters.offset,$__LIMIT,$current_user.is_logged_in,$current_user.contentobject.name)}
+        <div class="container-fluid main_cage row_list_box margin-bottom">
+            <div class="row">
+                {foreach $__ADULTS|extract($view_parameters.offset,$__LIMIT) as $__SINGLE_ADULT}
```

Comment:
la paginazione fatta così è solo a scopi grafici non solleva il db dalla fetch di tutti gli adulti in ogni pagina

Reply:
Perdona l'ignoranza, ma la fetch viene fatta ogni volta ad ogni cambio pagina se il template non ha cache visuale e il blocco di cache viene elaborato, se non ha cache già impostata, visto che non c'è modo  di pre-calcolare i vari elementi in un array da mantenere da qualche parte.
Spiegato in altre parole, se si fosse in symfony si fa la query(con le API del kernel) una volta sola, da li si ricava tutto e per le chiamate successive, la query non si fa più perchè si va a prendere dalla cache tutto il necessario senza ri-eseguire la query.